src/components/signup/ImagePicker/ImagePicker.tsx

```diff
@@ -0,0 +1,65 @@
+import { useRef, useState } from "react";
+import { useFormContext } from "react-hook-form";
+
+export interface IImagePickerProps {
+  name: string;
+}
+
+const ImagePicker = ({ name }: IImagePickerProps) => {
+  const { register, setValue } = useFormContext();
+
+  const [pickedImage, setPickedImage] = useState<string | null>(null);
+
+  const imageInput = useRef<HTMLInputElement>(null);
+
+  const buttonClickHandler = () => {
+    imageInput.current?.click();
+  };
+
+  const handleImageChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+
+    if (!file) {
+      setPickedImage(null);
+      setValue(name, null);
+      return;
+    }
+
+    const fileReader = new FileReader();
+
+    fileReader.onload = () => {
+      if (typeof fileReader.result === "string") {
+        setPickedImage(fileReader.result);
+      }
+    };
+
+    fileReader.readAsDataURL(file);
+
+    setValue(name, file);
+  };
+
+  return (
+    <div>
```

Comment:
지금 디자인된 프로필 설정 이미지 픽커는 '이미지 변경' 버튼은 없고, 이미지 자체를 선택해서 프로필을 변경하는 것처럼 보입니다.
아직 디자인을 적용하지 않은 컴포넌트 이겠지만, div 보다는 label로 태그를 바꾸면 useRef를 사용하지 않아도 될 것 같습니다.

Reply:
바쁘신 와중에 리뷰 감사합니다! 🙇

### **Image Picker** 구조
제가 만든 **Image Picker**는 이미지 미리보기 영역에서 absolute한 위치에 **버튼**을 두고,
이를 클릭해 이미지를 변경하는 구조로 설계했습니다.
(추후 퍼블리싱 작업에서는 `button`대신`img`로 대체할 계획입니다.)

혹시 지원님께서 말씀하신 `디자인된 프로필 설정 이미지 픽커`는 어떤 디자인을 말씀 하시는 건지 알 수 있을까요? 
- - -

### `ref` 대신 `label` 사용
아래 리뷰 내용과 관련이 있는데요,

`ref`를 사용해 클릭 이벤트를 트리거하게 되면 이미지 선택 여부와 관계없이
`input` 요소의 `onChange` 에 할당된 함수가 실행됩니다.

반면, `label`의 `htmlFor`속성과 `input` 요소의 `id`를 매칭하는 방식으로 구현하면,
이미지를 선택했을 때만 `onChange` 함수가 실행되는 걸로 알고 있습니다!
(혹시 제가 잘못 알고 있는 부분이 있다면 말씀 부탁드립니다! 🙏 )

그래서, `onChange`에 할당된 함수에
```typescript
if(!file) {
...
}
```
같은 검증 로직을 추가했는데요,

이 부분에서 **이미지를 선택하지 않을 경우 기본 이미지로 돌아갈지,**
**아니면 기존 이미지를 유지할지**에 대해서는 기획 및 디자인팀과 추가 논의가 필요할 것 같습니다!

- - -

#### 제가 참고한 Image Picker 디자인입니다.

<img width="364" alt="image" src="https://github.com/user-attachments/assets/e8b80da9-72c4-44ff-b710-59d968950494" />

<img width="425" alt="image" src="https://github.com/user-attachments/assets/ff4700a2-ec8c-4918-bfad-5cbdfcfc0f45" />

Reply:
제가 본 디자인도 같은 디자인인 것 같습니다. 
label을 사용할 경우 이미지 유무와 관계없이 onChange 함수를 실행할 수 있습니다.  
따라서 위 디자인과 같이 구현한다면 삭제버튼만 추가해서 삭제 이벤트 핸들러에 
위 코드에서 file이 없을 때 코드를 넣으면 될 것 같습니다.

```suggestion
<label htmlFor={name}>
        {pickedImage ? (
          <img src={pickedImage} alt={`${name} preview`} />
        ) : (
          <p>기본 이미지</p>
        )}
        <input
          hidden
          id={name}
          type="file"
          {...register(name)}
          ref={imageInput}
          onChange={handleImageChange}
        />
        {/* 아이콘 버튼 */}
        {/* <AddPhotoIcon /> */}
</label>
```

Reply:
늦은 시간에 리뷰 감사합니다 🙇

저는 `label`과 `input` 태그의 사용법은 아래처럼만 사용하던 터라,
`input`을 `label`로 감싸는 방법은 생각해보지 못했네요.

```html
<label for=""></label>
<input />
```

덕분에 새로운 방식도 알게 되었어요.
이렇게 하니까 정말 깔끔하게 동작하네요! 감사합니다. 👍👍